tiny/src/ui.rs

```diff
@@ -7,16 +7,78 @@ use crate::cmd::{parse_cmd, CmdArgs, ParseCmdResult};
 use crate::config;
 use futures::stream::StreamExt;
 use libtiny_client::Client;
-use libtiny_ui::{MsgSource, MsgTarget, UI};
+use libtiny_common::{ChanNameRef, MsgSource, MsgTarget, TabStyle};
+use libtiny_logger::Logger;
+use libtiny_tui::TUI;
 use std::path::{Path, PathBuf};
+use time::Tm;
 use tokio::sync::mpsc;
 
+macro_rules! delegate {
+    ( $name:ident ( $( $x:ident: $t:ty, )* )) => {
+        pub(crate) fn $name(&self, $($x: $t,)*) {
+            self.ui.$name( $( $x, )* );
+            if let Some(logger) = &self.logger {
+                logger.$name( $( $x, )* );
+            }
+        }
+    }
+}
+
+#[derive(Clone)]
+pub(crate) struct TinyUI {
```

Comment:
Let's call this `UI`; it's shorter and there isn't any ambiguity about what's UI is this.